confdb/aclocal_modules.m4

```diff
@@ -77,9 +77,11 @@ AC_DEFUN([PAC_CONFIG_HWLOC],[
     if test "$with_hwloc" = "embedded" ; then
         m4_if(hwloc_embedded_dir, [modules/hwloc], [
             dnl ---- the main MPICH configure ----
-            PAC_CONFIG_HWLOC_EMBEDDED([$VISIBILITY_CFLAGS])
-            hwlocsrcdir="${main_top_builddir}/modules/hwloc"
             hwloclib="${main_top_builddir}/modules/hwloc/hwloc/libhwloc_embedded.la"
+            if test ! -e "$hwloclib" ; then
+                hwlocsrcdir="${main_top_builddir}/modules/hwloc"
+                PAC_CONFIG_HWLOC_EMBEDDED([$VISIBILITY_CFLAGS])
+            fi
```

Comment:
So when not using prebuilt modules, would this mean the modules would not get reconfigured after they are built once? That gives me some concern users could run into this and not know what is going on.

Reply:
Yes.

Reply:
This is my biggest concern. To me, it violates the principle of least astonishment if a user reconfigures the build and it silently skips the embedded modules, even if explicitly used. Is there some way we can special case the prebuilt module so that it doesn't interfere with a regular build?

Reply:
It *does not* skip the embedded modules, it just skips the `configure` and use the already built `libhwloc-embedded.la`. It does change the old behavior that when user want to *reconfigure* libhwloc, they need `make clean` or remove the `libhwloc-embedded.la` first.

If that slight change is undesiable, I guess we can embed flag file in the prebuilt tarball so this skipping of configure only happens when we use prebuilt tarballs.

Reply:
> If that slight change is undesiable, I guess we can embed flag file in the prebuilt tarball so this skipping of configure only happens when we use prebuilt tarballs.

That sounds like a good alternative.

Reply:
It seems like this feedback was ignored. This is still a NACK from me.

Reply:
Ahh, sorry about that. Let me fix it.

Reply:
What I am concerned about is a scenario where a user configures/builds MPICH, then later decides to configure/build again with different options:
```
./configure --prefix=foo && make install
```
```
./configure --prefix=bar --with-cuda && make install
```
The second configuration will use the already built submodules without CUDA support.